test/e2e/service.go

```diff
@@ -72,9 +72,19 @@ var _ = framework.KubeDescribe("Services", func() {
 	f := framework.NewDefaultFramework("services")
 
 	var c *client.Client
+	var gceController *GCEIngressController
 
 	BeforeEach(func() {
 		c = f.Client
+		By("Initializing gce controller")
+		gceController = &GCEIngressController{ns: f.Namespace.Name, Project: framework.TestContext.CloudConfig.ProjectID, c: c}
+		gceController.init()
+	})
+
+	// Platform specific cleanup
+	AfterEach(func() {
+		By("Cleaning up cloud resources")
+		cleanupGCE(gceController)
```

Comment:
you essentially need to do this: https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/delete-stranded-load-balancers.sh#L25

But figure out the expected resource names from the UIDs of the Services created for the particular test, and only delete those resources.